cmd/fyne/internal/commands/build.go

```diff
@@ -46,37 +50,43 @@ func checkGoVersion(runner runner, versionConstraint *version.ConstraintGroup) e
 	return checkVersion(string(goVersion), versionConstraint)
 }
 
-func (b *builder) build() error {
+func (b *builder) build() ([]string, error) {
 	var versionConstraint *version.ConstraintGroup
 
-	if b.runner == nil {
-		b.runner = newCommand("go")
-	}
-
 	goos := b.os
 	if goos == "" {
 		goos = targetOS()
 	}
 
+	if b.runner == nil {
+		if goos != "gopherjs" {
+			b.runner = newCommand("go")
+		} else {
+			b.runner = newCommand("gopherjs")
+		}
+	}
+
 	args := []string{"build"}
 	env := os.Environ()
 
-	if goos != "wasm" {
+	if !isWeb(goos) {
 		env = append(env, "CGO_ENABLED=1") // in case someone is trying to cross-compile...
 	}
 
 	if goos == "darwin" {
 		env = append(env, "CGO_CFLAGS=-mmacosx-version-min=10.11", "CGO_LDFLAGS=-mmacosx-version-min=10.11")
 	}
 
-	if goos == "windows" {
-		if b.release {
-			args = append(args, "-ldflags", "-s -w -H=windowsgui")
-		} else {
-			args = append(args, "-ldflags", "-H=windowsgui")
+	if !isWeb(goos) {
```

Comment:
There is an identical check 8 lines above, could it be simplified by combining?
Most likely by moving the `if goos == "darwin" ` into the `if !isWeb` :)